test/differentials/composite.jl

```diff
@@ -75,6 +74,22 @@ end
         @test collect(Composite{Tuple{Float64,}}(2.0)) == [2.0]
     end
 
+    @testset "reverse" begin
+        t = (1, 2, "something")
+        tr = reverse(t)
+        c = Composite{typeof(t), typeof(t)}(t)
+        cr = Composite{typeof(tr), typeof(tr)}(tr)
+        @test reverse(c) == cr
```

Comment:
We should avoid being smart in tests and instead be obvious.
```suggestion
        c = Composite{Tuple{Int, Int, String}}(1, 2, "something")
        cr = Composite{Tuple{String, Int, Int}}("something", 2, 1)
        @test reverse(c) === cr
```
also === should make sure this only passes if exactly same including type

Reply:
Thanks, I need to learn to be more explicit in tests